xtrlock.c

```diff
@@ -166,48 +236,142 @@ int main(int argc, char **argv){
     exit(1);
   }
 
-  for (;;) {
+  if((!blank_screen) && (blink_delay!= 0)){/*blink to indicate a successful lock*/
```

Comment:
You don't need to put parentheses around ```!blank_screen``` or ```blink_delay!=0```. Both ```!``` and ```!=``` have precedence above ```&&```.